packages/path-provider/example/lib/main.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+void main() {
+  runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Path Provider',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(title: 'Path Provider'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key, this.title}) : super(key: key);
+  final String title;
+
+  @override
+  _MyHomePageState createState() => new _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  Future<Directory> _tempDirectory;
+  Future<Directory> _appDocumentsDirectory;
+
+  void _requestTempDirectory() {
+    setState(() {
+      _tempDirectory = getTemporaryDirectory();
+    });
+  }
+
+  Widget _buildDirectory(
+      BuildContext context, AsyncSnapshot<Directory> snapshot) {
+    Text text;
+
+    if (snapshot.connectionState == ConnectionState.done) {
+      if (snapshot.hasError) {
+        text = new Text('Error: ${snapshot.error}');
+      } else if (snapshot.hasData) {
+        text =  new Text('path: ${snapshot.data.path}');
+      } else {
+        text = new Text('path unavailable');
+      }
+    } else {
+      text = new Text('');
+    }
```

Comment:
`new Text(foo)` should be `const Text(foo)` when `foo` is a constant string.

As an alternatively to the `else` clause in lines 58-60, you could declare `Text text = const Text('');` in line 48.